design/ux/Obscuro_Gateway.md

```diff
@@ -0,0 +1,264 @@
+
+
+## High level UX
+
+The key reason for the OG is to allow implementing a 3-click user onboarding process.
+
+See below for a proposed flow:
+![UX diagram](resources/og_ux.png)
+
+```plantuml
+@startuml
+'https://plantuml.com/sequence-diagram
+
+autonumber
+
+actor "Alice's Browser" as Alice
+participant MetaMask as MM
+participant "https://gateway.obscuro.network/v1" as OG
+participant "https://obscuro.network" as ON
+
+group First click
+    Alice -> ON: Join Obscuro
+    ON --> Alice: Redirect to OG
+    note right
+        The point of this sequence
+        is to call the "Create User"
+        endpoint on the OG
+    end note
+    Alice -> OG: Create User\n(ajax call behind the scenes)
+    OG -> OG: Generate and record VK\nand record against UserId
+    note right
+        The UserId is the hash of
+         the Public Key of the VK
+    end note
+    OG -> Alice: Send UserId
+    Alice -> MM: Automatically add "Obscuro" network with RPC\n"https://gateway.obscuro.network/v1?u=$UserId"
+end
+
+group Second click
+    Alice -> MM : Connect wallet and switch to Obscuro
+end
+
+group Third click
+    Alice -> MM: Automatically open MM with request to \nsign over "Register $UserId for $Acct"
+    note right
+        This text will be sent as is
+        accompanied by the signature and
+        will be parsed inside the enclave
+    end note
+    Alice -> MM : Confirm signature
+end
+
+Alice -> OG: All further Obscuro interactions will be to\nhttps://gateway.obscuro.network/v1?u=$UserId
+
+@enduml
+```
+
+The onboarding should be done in 3 clicks.
+1. The user goes to a website (like "obscuro.network"), where she clicks "Join Obscuro". This will add a network to their wallet.
+2. User connects the wallet to the page.
+3. In the wallet popup, the user has to sign over a message: "Register $UserId for $ACCT"
+
+##### Click 1
+1. Behind the scenes, a js functions calls "gateway.obscuro.network/v1/join" where it will generate a VK and send back the hash of the Public key. This is the "UserId" 
+2. After receiving the UserId, the js function will add a new network to the wallet.
+The RPC URL of the new Obscuro network will include the userid: "https://gateway.obscuro.network/v1?u=$UserId". 
+Notice that the UserId has to be included as a query parameter because it must be encrypted by https, as it is secret.
+
+##### Click 2
+After these actions are complete, the same page will now ask the user to connect the wallet and switch to Obscuro. 
+Automatically the page will open metamask and ask the user to sign over a text "Register $UserId for $ACCT", where ACCT is the current account selected in metamask.
+
+##### Click 3
+Once signed, this will be submitted in the background to: "https://gateway.obscuro.network/v1?u=$UserId&action=register"
+
+
+Note: Any further accounts will be registered similarly for the same UserId.
```

Comment:
Likely covered in the WE, but throwing in as you might find it significant. Should multiple address registration for the same UserId be detailed ?